src/grpc/infer_handler.h

```diff
@@ -627,22 +627,29 @@ class InferHandlerState {
         ::grpc::ServerCompletionQueue* cq, const uint64_t unique_id = 0)
         : cq_(cq), unique_id_(unique_id), ongoing_requests_(0),
           step_(Steps::START), finish_ok_(true), ongoing_write_(false),
-          received_notification_(false)
+          notify_state_(nullptr), received_notification_(false)
     {
       ctx_.reset(new ::grpc::ServerContext());
       responder_.reset(new ServerResponderType(ctx_.get()));
     }
 
+    ~Context()
+    {
+      if (notify_state_ != nullptr) {
+        delete notify_state_;
+      }
+    }
+
     void SetCompressionLevel(grpc_compression_level compression_level)
     {
       ctx_->set_compression_level(compression_level);
     }
 
     void GrpcContextAsyncNotifyWhenDone(InferHandlerStateType* state)
     {
-      InferHandlerStateType* wrapped_state =
+      notify_state_ =
```

Comment:
If there is only one notify state per context, can this just be a `unique_ptr` or something to auto clean up when context is destructed?

Reply:
That's a good idea!